tensorboard/plugins/npmi/add_embeddings.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow as tf
+from absl import app
+from absl import flags
+from tensorboard.plugins.npmi import summary
+from tensorboard.plugins.npmi import metadata
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string(
+    "log_path", None, "Log file to extend by embedding data.",
+)
+flags.DEFINE_string(
+    "out_path", None, "Where to write the new log file.",
+)
+flags.DEFINE_string(
+    "embeddings_path",
+    None,
+    "Path to the numpy file containing the embeddings.",
+)
+flags.mark_flag_as_required("log_path")
+flags.mark_flag_as_required("out_path")
+flags.mark_flag_as_required("embeddings_path")
+
+
+def add_embeddings(log_path, out_path, embeddings_path):
+    with open(embeddings_path, "rb") as f:
+        embeddings = np.load(f)
+    event_file = log_path
+    embeddings_tensor = tf.convert_to_tensor(embeddings)
+    writer = tf.compat.v2.summary.create_file_writer(out_path)
+    with writer.as_default():
+        summary.npmi_embeddings(embeddings_tensor, 1)
+        for summ in tf.compat.v1.train.summary_iterator(event_file):
```

Comment:
The point of the loop here is essentially just to copy existing NPMI events from the event file, right?  Is that strictly necessary?  TensorBoard is designed to consume multiple event files, so having the embeddings in one event file and the other 3 NPMI summaries in the other one shouldn't really be observable by the plugin.

So it seems a bit simpler if the tool just takes an embedding .npy file and converts it into an event file.  If you really need a way to concatenate the event files together perhaps that can be done as a separate utility that's more generic.

In particular, as written, this tool also has the effect of stripping non-NPMI summaries from the input event file which seems potentially surprising (and isn't necessarily suggested by the current flag documentation).

Reply:
Completely agree, and was a misconception from my side. I changed that now.